week7/cooper_client_react/01_cooper_calculator.md

```diff
@@ -0,0 +1,398 @@
+
+***"++Remember to commit often++"***
+
+Lets start with cleaning up `App.js` aka the App component. When that is done we want to add the inputs and selectors that we need to make this feature test to go green.
+The App component should look like this:
+
+```js
+import React, { Component } from 'react';
+import './App.css';
+
+class App extends Component {
+ render() {
+    return (
+      <div className="App">
+        <div>
+          <label>Distance</label>
+          <input id="distance"></input>
+        </div>
+
+        <select id="gender">
+          <option value="female">Female</option>
+          <option value="male">Male</option>
+        </select>
+
+        <div>
+          <label>Age</label>
+          <input id="age"></input>
+        </div>
+      </div>
+    );
+  }
+}
+
+export default App;
+```
+
+When running the test now, we should not get any errors `beforeEach` block that we have in our feature test. It now finds all the fields and input the correct values that we have set in the test.
+
+So now it is time for us to display the result. This is the time for us to do some component testing because we need to create a new component that returns the result to us.
+
+Run this in the terminal:
+
+```shell
+$ npm i -D enzyme enzyme-adapter-react-16 react-scripts react-test-renderer sinon
+$ touch src/setupTests.js
+$ mkdir src/__tests__
+$ touch src/__tests__/DisplayCooperResult.test.js
+```
+
+Add this to the files we added:
+
+```js
+// setupTests.js
+import { configure } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+
+configure({ adapter: new Adapter() }); 
+```
+
+```js
+// DisplayCooperResult.test.js
+import React from 'react';
+import { mount, shallow } from 'enzyme';
+
+import DisplayCooperResult from '../DisplayCooperResult';
+
+describe('<DisplatCooperResult />', () => {
+  it('evaluates the correct result for female/poor', () => {
+    const describedComponent = shallow(<DisplayCooperResult distance="1000" gender="female" age="23"/>);
+    const response = <p>Result: Poor</p>
+    expect(describedComponent.contains(response)).toEqual(true)
+  })
+
+  it('evaluates the correct result for female/average', () => {
+    const describedComponent = shallow(<DisplayCooperResult distance="2000" gender="female" age="23"/>);
+    const response = <p>Result: Average</p>
+    expect(describedComponent.contains(response)).toEqual(true)
+  })
+})
+```
+After doing the BMI Calculator we know how to send in props to a child component. We send in `distance`, `gender` and `age`. Then we expect to get a result that the component renders.
+
+If you run `npm run test` now it will complain that it cant find the `DisplayCooperResult` module. That is not that weird considering we dont have that component. Lets create it:
+
+`$ mkdir src/Components`
+
+`$ touch src/Components/DisplayCooperResult.js`
+
+And add this to that file:
```

Comment:
I would like us to be more clear about what we want to do. Like: "To start with, add a stateful/class component with a `render` method`. We will expand it with more functionality as we move ahead" or something like that.

Reply:
Done